hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/filters/With.java

```diff
@@ -119,6 +122,39 @@ public static PropertyValues propertyValues(String name, Object... values) {
         return new PropertyValues(name, values);
     }
 
+    /**
+     * Checks whether there is data on the provided relative path. This filter is only really applicable
+     * on the {@link org.hawkular.inventory.api.model.DataEntity}. I.e. the relative path is always resolved
```

Comment:
> This filter is only really applicable on the {@link org.hawkular.inventory.api.model.DataEntity}

So it's similar as the `RelationWith` thing, right (i.e. not completely type safe)? What about having `WithData` class?